### PR TITLE
fix(consumer): goroutine leak when shutdown

### LINKED ISSUE #285: [Native] goroutine leak after  pushConsumer  new and Shutdown frequently
**BUG REPORT**  
**Please add the branch name [Native]/[Master] at the header of the Isssue title.**

1. Please describe the issue you observed:

    - What did you do (The steps to reproduce)?
    goroutine grows more and more when create and destroy pushConsumer frequently.

    - What did you expect to see?
    gotoutine should keep smooth

    - What did you see instead?

2. Please tell us about your environment:

     - What is your OS?

     - What is your client version?

     - What is your RocketMQ version?

3. Other information (e.g. detailed explanation, logs, related issues, suggestions on how to fix, etc):

### PR DESCRIPTION
**Pull request by xujianhai666**
- add done chan for lifecycle close

Closes #285

### CODE REVIEW

**Comment by ShannonDing** [comment]
internal/remote/future.go:25:1: invalid import path: ``

**Comment by xujianhai666** [comment]
> internal/remote/future.go:25:1: invalid import path: ``

@ShannonDing done

File: consumer/push_consumer.go [hunk 1]
```diff
@@ -175,6 +191,7 @@ func (pc *pushConsumer) Start() error {
 
 func (pc *pushConsumer) Shutdown() error {
 	pc.lockTicker.Stop()
+	close(pc.done)
```
Comment (wenfengwang) [inline consumer/push_consumer.go]: it is better that add a `sync.Once` to wrap the block to avoiding double call

Comment (xujianhai666) [reply consumer/push_consumer.go]: @wenfengwang done

Comment (xujianhai666) [reply consumer/push_consumer.go]: @wenfengwang done

**Pull request merged**